doc/dev/developer_guide/debugging-gdb.rst

```diff
@@ -0,0 +1,43 @@
+GDB - The GNU Project Debugger
+==============================
+
+`The GNU Project Debugger (GDB) <https://www.sourceware.org/gdb>`_ is
+a powerful tool that allows you to analyze the execution flow
+of a process.
+GDB can help to find bugs, uncover crash errors or track the
+source code during execution of a development cluster.
+It can also be used to debug Teuthology tests runs.
```

Comment:
s/tests runs/test runs/

Reply:
Ah, missed this @zdover23. Thanks, I'll open a small PR for it.